docs/user-guide/reduction-workflow-guidelines.md

```diff
@@ -0,0 +1,139 @@
+
+**Reason**
+- Required for polarization analysis, which wraps a base workflow.
+- Required for subsequent filtering, unless part of the workflow.
+
+**Note**
+There should be a workflow parameters (flag) to select whether to return event data or not.
+
+### S.2: Split loading and handling of monitors and detectors from loading of auxiliary data and metadata
+
+**Reason**
+- Allows for more efficient parallelism and reduction in memory use.
+- Avoids loading large data into memory that is not needed for the reduction.
+- Avoids keeping large data alive in memory if output metadata extraction depends on auxiliary input data or input metadata.
+
+### S.3: Avoid dependencies of output metadata on large data
+
+**Reason**
+Adding dependencies on large data to the output metadata extraction may lead to large data being kept alive in memory.
+
+**Note**
+Most of this is avoid by following S.2.
+A bad exampleAvoid would be to write to total raw counts to the output metadata, as this would require keeping the large data alive in memory, unless it is ensured that the task runs early.
+
+### S.4: Handle single- and double-precision transparently
+
+**Reason**
+Single-precision may be sufficient for most data.
+By writing workflows transparently for single- and double-precision, we avoid future changes if we either want to use single-precision for performance reasons or double-precision for accuracy reasons.
+
+**Note**
+This affects coordinates and data values independently.
+- If input counts are single-precision, the reduced intensity should be single-precision, and equivalently for double-precision.
+- If input coordinates are single-precision, derived coordinates should be single-precision, and equivalently for double-precision.
+
+**Note**
+This will allow for changing the precision of the entire workflow by choosing a precision when loading the input data.
+
+**Example**
+- If time-of-flight is single-precision, wavelength and momentum transfer should be single-precision.
+- In counts are single-precision, reduced intensity should be single-precision.
+
+### S.5: Switches to double-precision shall be deliberate, explicit, and documented
+
+**Reason**
+Some workflows may require switching to double-precision at a certain point in the workflow.
+This should be a deliberate choice, and the reason for the switch should be documented.
+
+### S.6: Propagation of uncertainties in broadcast operations should support "drop" and "upper-bound" strategies, "upper-bound" shall be the default
+
+**Reason**
+Unless explicitly computed, the exact propagation of uncertainties in broadcast operations is not well-defined.
+Dropping uncertainties is not desireable in general, as it may lead to underestimation of the uncertainties, but we realize that the upper-bound approach may not be suitable in all cases.
+We should therefore support two strategies, "drop" and "upper-bound", and "upper-bound" should be the default.
```

Comment:
Is this really something the user should toggle? How can the upper-bound strategy be valid for some data but not other for the same workflow?

Reply:
I am pretty sure that at least the IDS or IS will want to toggle this.